easybuild/tools/filetools.py

```diff
@@ -2980,3 +2987,81 @@ def create_unused_dir(parent_folder, name):
     set_gid_sticky_bits(path, recursive=True)
 
     return path
+
+
+def get_first_nonexisting_parent(path):
+    path = os.path.abspath(path)
+
+    first_nonexisting_parent = None
+    while not os.path.exists(path):
+        first_nonexisting_parent = path
+        path = os.path.dirname(path)
+
+    return first_nonexisting_parent
+
+
+def _apply_suffix(paths, number):
+    suffix = '_' + str(number)
+
+    final_paths = []
+    first_nonexisting_parents = []
+    for p in paths:
+        final_paths.append(p + suffix)
+        first_nonexisting_parents.append(p)
+
+    return final_paths, first_nonexisting_parents
+
+
+def _clean_directories(paths):
+    for p in paths:
+        shutil.rmtree(p, ignore_errors=True)
+
+
+def create_unused_dirs(paths, index_upper_bound=10000):
+    """
+    Create directories with given paths, including the parent directories
+    When a directory in the same path for any of the path  in the path list already exists, then the suffix '_<i>' is
+    appended for i=0..(index_upper_bound-1) until an index is found so that all required path as unused. All created
+    directories have the same postfix.
```

Comment:
To be consistent:
```suggestion
    directories have the same suffix.
```

Maybe put the "All created" on the same line because that is the most important part of the function, isn't it?